src/chart_types/partition_chart/state/selectors/tree.ts

```diff
@@ -20,33 +20,43 @@
 import createCachedSelector from 're-reselect';
 
 import { ChartTypes } from '../../..';
+import { CategoryKey } from '../../../../common/category';
 import { SpecTypes } from '../../../../specs';
-import { GlobalChartState } from '../../../../state/chart_state';
+import { GlobalChartState, SpecList } from '../../../../state/chart_state';
 import { getSpecsFromStore } from '../../../../state/utils';
 import { configMetadata } from '../../layout/config/config';
-import { childOrders, HierarchyOfArrays, HIERARCHY_ROOT_KEY } from '../../layout/utils/group_by_rollup';
+import { childOrders, HIERARCHY_ROOT_KEY, HierarchyOfArrays } from '../../layout/utils/group_by_rollup';
 import { getHierarchyOfArrays } from '../../layout/viewmodel/hierarchy_of_arrays';
 import { isSunburst, isTreemap } from '../../layout/viewmodel/viewmodel';
 import { PartitionSpec } from '../../specs';
 
 const getSpecs = (state: GlobalChartState) => state.specs;
 
+const getDrilldownSelection = (state: GlobalChartState) => state.interactions.drilldown || [];
+
 /** @internal */
 export const getTree = createCachedSelector(
-  [getSpecs],
-  (specs): HierarchyOfArrays => {
+  [getSpecs, getDrilldownSelection],
+  (specs: SpecList, drilldownSelection: CategoryKey[]): HierarchyOfArrays => {
```

Comment:
This typing is inferred from the selectors.
```suggestion
  (specs, drilldownSelection): HierarchyOfArrays => {
```